### ISSUE TITLE
Integrating with TinyGSM

### ISSUE THREAD

**Issue by woodlist** [shurillu/CTBot#83]
I did not find how to put an "enhancement" label to this record.
It's would be perfect to get used this library not only by wifi connection, but also with customized connection to Internet, like by GSM module.

**Comment by shurillu** [comment]
Hello woodlist,
   thank you for using the library!
I put the "enhancement" label to this thread for you ;-)
Basically, all the connection stuff is located on the CTBotSecureConnection class. It use the WiFiClientSecure class from ESP8266/ESP32 toolchain.
It seems to be an interesting thread (BTW I'm thinking about using LORA mesh networks and WiFi/LORA gateway in order to use Telegram bots with LORA... but it's only an idea for now!).
So please, let me know what are you thinking on ;-)

Cheers

Stefano

**Comment by woodlist** [comment]
Dear Stefano, thanks quick reply.
Of course, the LORA mesh network coverage will be an great gift for users, but since lower distance range compared with GSM network, the GSM connection to telegram bot is much actual subject. I recommend the the TinyGSM as integration library, cause of being well maintained and widely used.
Best Regards.

**Comment by woodlist** [comment]
> Love the idea. I used T-Call Sim800L and I used TinyGSM+Blynk. It worked perfectly for me. TinyGSM and Blynk are closely integrated maybe because TinyGSM owner also Blynk founder. @woodlist have you tried TinyGSM with Telegram Bot?

Yes, I have successful experience of using the TinyGSM with [https://github.com/witnessmenow/Universal-Arduino-Telegram-Bot](url)
If given library author will be so honest to build beneath kind of constructor, as has the Universal Telegram Bot:
`WiFiClientSecure secured_client;
UniversalTelegramBot bot(BOT_TOKEN, secured_client);`
the task will be done easily.
